File: java/src/test/java/antipatterns/BlogTests.java
package antipatterns;

import blog.Article;
import lombok.val;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class BlogTests {
    @Test
    void it_should_return_a_Right_for_valid_comment() {
        val article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        val result = article.addComment("Amazing article !!!", "Pablo Escobar");

        assertThat(result.isRight()).isTrue();
    }

    @Test
    void it_should_add_a_comment_with_the_given_text() {
        val article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        val text = "Amazing article !!!";
        val result = article.addComment(text, "Pablo Escobar");

        assertThat(result.get().getComments())
                .hasSize(1)
                .anyMatch(comment -> comment.getText().equals(text));
    }

    @Test
    void it_should_add_a_comment_with_the_given_author() {
        val article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        val author = "Pablo Escobar";
        val result = article.addComment("Amazing article !!!", author);

        assertThat(result.get().getComments())
                .hasSize(1)
                .anyMatch(comment -> comment.getAuthor().equals(author));
    }

    @Test
    void it_should_add_a_comment_with_the_date_of_the_day() {
        val article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        val result = article.addComment("Amazing article !!!", "Pablo Escobar");
    }

    @Test
    void it_should_return_a_Left_when_adding_existing_comment() {
        val article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        val result =
                article.addComment("Amazing article !!!", "Pablo Escobar")
                        .map(a -> a.addComment("Amazing article !!!", "Pablo Escobar"))
                        .flatMap(r -> r);

        assertThat(result.isLeft()).isTrue();
    }
}

File: csharp/CleanTests.Tests/AntiPatterns/BlogTests.cs
using CleanTests.Blog;
using FluentAssertions;
using Xunit;

namespace CleanTests.Tests.AntiPatterns;

public class BlogTests
{
    [Fact]
    public void It_Should_Return_a_Right_For_Valid_Comment()
    {
        var article = new Article(
            "Lorem Ipsum",
            "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        var result = article.AddComment("Amazing article !!!", "Pablo Escobar");

        result.IsRight.Should().BeTrue();
    }

    [Fact]
    public void It_Should_Add_A_Comment_With_The_Given_Text()
    {
        var article = new Article(
            "Lorem Ipsum",
            "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        const string text = "Amazing article !!!";
        var result = article.AddComment(text, "Pablo Escobar");

        result.RightUnsafe()
            .Comments.Should()
            .HaveCount(1)
            .And
            .Satisfy(comment => comment.Text == text);
    }

    [Fact]
    public void It_Should_Add_A_Comment_With_The_Given_Author()
    {
        var article = new Article(
            "Lorem Ipsum",
            "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        const string author = "Pablo Escobar";
        var result = article.AddComment("Amazing article !!!", author);

        result.RightUnsafe()
            .Comments.Should()
            .HaveCount(1)
            .And
            .Satisfy(comment => comment.Author == author);
    }

    [Fact]
    public void It_Should_Add_A_Comment_With_The_Date_Of_The_Day()
    {
        var article = new Article(
            "Lorem Ipsum",
            "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        var result = article.AddComment("Amazing article !!!", "Pablo Escobar");
    }

    [Fact]
    public void It_Should_Add_A_Left_When_Adding_Existing_Comment()
    {
        var article = new Article(
            "Lorem Ipsum",
            "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
        );

        var result = article.AddComment("Amazing article !!!", "Pablo Escobar")
            .Map(a => a.AddComment("Amazing article !!!", "Pablo Escobar"))
            .Flatten();

        result.IsLeft.Should().BeTrue();
    }
}